File: radius.py
import numpy as np
#from plot import *
#from update import *
#from radius import *
from start import *
  
def radius(n, L, k, m, model, prop_openminded, position):
    r = 0
    r_l = 0
    r_r = 0
    if model == 'Homogeneous':
        r = np.ones(n)
    elif model == 'Proportion open':       
        r = np.zeros(n)
        while sum(r) < np.round(n*prop_openminded, 0):
            index = np.random.choice(n,1)
            r[index] = 1
    elif model == 'Open right':
        r = k*position + m
    elif model == 'Open center':
        l_half = [i for i, e in enumerate(position) if e <= L/2]
        if l_half == []:
            r = -k*(position - L/2) + k*L/2 + m
        elif len(l_half) == len(position):
            r = k*position + m
        else:    
            r_l = k*position[0:max(l_half) + 1] + m
            r_r = -k*(position[max(l_half)+1:] - L/2) + k*L/2 + m
            r = np.append(r_l,r_r) 
    return r
#radius(10, 0, 1, 3, 'homo', 1, 1)

File: radius.jl
function radius(n, L, model, dist, prop_o, k, m)
    opinions = start_pos(dist, n, L)
    if model == "homo"
        r = ones(1, n)
    elseif model == "prop"
        r = collect(1:n) - collect(1:n)
        while sum(r) < round(n*prop_o, digits = 0)
            r[rand(1:n)] = Int(1)
        end
    elseif model == "open r"
        r = k .* opinions .+ m
    elseif model == "open c"
        index = 0
        for i in 1:length(opinions)
            if opinions[i] <= maximum(opinions)/2
                index = index + 1
            else
                break
            end
        end
        radius_left = k .* opinions[1:index]
        radius_right = -k .* (opinions[index+1:end] .- L/2) .+ k.*L/2
        r = [radius_left; radius_right]
        #r = r./L .+ m # Scale
        r = r ./ L
        r = r .+ m
    else
        println("Wrong model")
        r = []
    end
    return r
end
#radius(12, 5, "open r", "equi", 0.5, 1, 0)
